src/main/java/mineverse/Aust1n46/chat/utilities/Format.java

```diff
@@ -121,7 +212,7 @@ private static String convertPlaceholders(String s, JsonFormat format, Mineverse
 							hover.append(Format.FormatStringAll(st) + "\n");
 						}
 						final String hoverText;
-						if(!hover.isEmpty()) {
+						if(hover.length() != 0) {

```

Comment:
I know isEmpty() isn't available in older Java versions, but I don't think I should be forced to develop against older versions.  Java 17 is the new LTS.  The 4.0 dev branch uses Java 17 features as well, so it's a lot more than just this one line.